### DISCUSSION TITLE
Looking for a very basic Example: mp3 from SD-Card to I2S with custom pin-config

### DISCUSSION THREAD

**Discussion by FerFemNemBem** [pschatzmann/arduino-audio-tools#158] [Q&A] [answered]
Hi,
I'm new to C & Arduino but in most cases I can find examples that work for me. Now I'm struggeling with a very simple thing:
I want to play a single mp3-file from an SD-Card to I2S. Sounds simple (maybe it is) but I have no Idea how to configure the pins and define the file(name). The maybe most related example "player-sd-i2s". But this has a "startFilePath" and an "extension" only, not a filename itself defined. Further - where can I define my pin-config for I2S and the SPI? The following pseudocode-snippets hopefully explain what I want toi do and what I'm missing:
```
#include <Arduino.h>
#include <FS.h>
#include "AudioTools.h"
#include "AudioCodecs/CodecMP3Helix.h"

#define USE_HELIX 
#define USE_SDFAT

SPIClass spiSD(HSPI);
#define SD_CS 5

File mp3File;

#define LRCLK 25
#define BCLK 26
#define DIN 33

AudioSourceSdFat source(startFilePath, ext);
I2SStream i2s; **=> where to config my i2s pins?**
MP3DecoderHelix decoder;
AudioPlayer player(source, i2s, decoder);

void setup() 
{
  **=> Where to define the SPI-Pins for the "AudioSourceSdFat" like the spiSD.begin(...) method?**
  //spiSD.begin(14, 12, 13, SD_CS);
  //SD.begin(SD_CS, spiSD))
  //mp3File = SD.open("/Sound/test.mp3");
  ...
  player.begin();
}

void loop()
{
  player.copy();
}
```
Thank you!

Best regards,
FFNB

**Comment by pschatzmann** [comment, marked as answer]
I suggest that you take the example [streams-sd_mp3-i2s](https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-stream/streams-sd_mp3-i2s). 

The AudioPlayer is for playing many files, therefore you indicate just a directory. In addition please notice that this class is using the https://github.com/greiman/SdFat library!


For the pins I suggest that you read the chapter about [External-DAC](https://github.com/pschatzmann/arduino-audio-tools/wiki/External-DAC)

I also recommend that you have a look at the [class documentation](https://pschatzmann.github.io/arduino-audio-tools/html/annotated.html) where you find all the classes with their methods and the pubic attributes

**Reply by FerFemNemBem** [reply]
@pschatzmann 

You nailed it! This is exactly what I was looking for!
Thank you!

Best regards,
Jan
